File: examples/micronaut-app/src/test/java/ht/eyfout/example/http/ExampleStateScopeProvider.java
package ht.eyfout.example.http;

import ht.eyfout.junit.jupiter.api.http.HttpAPIStateScopeProvider;
import ht.eyfout.junit.jupiter.api.http.HttpAPIThenScope;

public class ExampleStateScopeProvider extends HttpAPIStateScopeProvider<ExampleGivenState, ExampleWhenScope, HttpAPIThenScope> {
    public static HttpAPIStateScopeProvider<ExampleGivenState, ExampleWhenScope, HttpAPIThenScope> INSTANCE = new ExampleStateScopeProvider();

    @Override
    public ExampleGivenState givenState() {
        return new ExampleGivenState();
    }

    @Override
    public ExampleWhenScope whenScope(ExampleGivenState givenState) {
        return new ExampleWhenScope(givenState);
    }
}

File: examples/micronaut-app/src/test/kotlin/ht/eyfout/example/http/ExampleStateScopeProvider.kt
package ht.eyfout.example.http

import ht.eyfout.example.client.DMVClient
import ht.eyfout.junit.jupiter.gherkin.api.http.HttpAPIStateScopeProvider
import ht.eyfout.junit.jupiter.gherkin.api.http.HttpAPIThenScope
import ht.eyfout.junit.jupiter.gherkin.api.http.HttpAPIWhenScope
import jakarta.inject.Singleton

@Singleton
class ExampleStateScopeProvider(private val client: DMVClient) :
    HttpAPIStateScopeProvider<ClientGivenState, HttpAPIWhenScope, HttpAPIThenScope>() {
    override fun givenState(): ClientGivenState {
        return ClientGivenState(client = client)
    }
}